python/ql/src/experimental/Classes/NamingConventionsClasses.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @name Misnamed class
+ * @description A class name that begins with a lowercase letter decreases readability.
+ * @kind problem
+ * @problem.severity recommendation
+ * @id py/misnamed-class
+ * @tags maintainability
+ */
+
+import python
+
+from Class c, string first_char
+where
+  c.inSource() and
+  first_char = c.getName().prefix(1) and
+  not first_char = first_char.toUpperCase() and
```

Comment:
I think it would help readability (and avoid repetition) if this was factored out into a separate predicate, for instance
```ql
predicate lower_case_class(File file, Class c) {
  file = c.getLocation().getFile()
  exists(string first_char | 
    first_char = c.getName().prefix(1) and
    not first_char = first_char.toUpperCase()
  )
}
```
(untested)

That way, you can reference this twice in the query.